src/pages/files.tsx

```diff
@@ -1,64 +1,383 @@
+
+          try {
+            await props.contentsManager.save(finalFileName, {
+              type: 'file',
+              format: isImage ? 'base64' : 'text',
+              content: finalContent
+            });
+
+            uploadedFiles.push({
+              id: fileId,
+              name: finalFileName,
+              size: file.size,
+              type: file.type,
+              lastModified: file.lastModified,
+              content
+            });
+          } catch (error) {
+            console.warn(`Upload skipped or failed for ${finalFileName}`, error);
+          }
+        }
+
+        if (uploadedFiles.length > 0) {
+          props.onFilesUploaded(uploadedFiles);
+        }
+      } finally {
+        props.onUploadEnd();
+      }
+    },
+    [props]
+  );
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files) {
+      handleFileSelect(e.target.files);
+    }
+  };
+
+  const triggerFileSelect = () => {
+    fileInputRef.current?.click();
+  };
+
+  // Expose the trigger function to the parent
+  React.useImperativeHandle(ref, () => ({
+    triggerFileSelect
+  }));
+
   return (
-    <button className="je-Tile">
+    <input
+      ref={fileInputRef}
+      type="file"
+      multiple
+      onChange={handleInputChange}
+      style={{ display: 'none' }}
+      accept=".png,.jpg,.jpeg,.csv,image/png,image/jpeg,text/csv"
+    />
+  );
+});
+
+FileUploader.displayName = 'FileUploader';
+
+/**
+ * This component displays a thumbnail for each uploaded image or file.
+ * We use the `getFileIcon` function to determine the icon based on file type.
+ */
+interface IFileThumbnailProps {
+  file: IUploadedFile;
+  onRemove: (fileId: string) => void;
+  contentsManager: Contents.IManager;
+}
+
+/**
+ * FileThumbnail component displays a thumbnail for an uploaded file.
+ * @param props - The properties for the FileThumbnail component, including the file to display and a callback for removing the file.
+ * @returns A JSX element representing the file thumbnail.
+ */
+function FileThumbnail(props: IFileThumbnailProps) {
+  const { file } = props;
+
+  const handleRemove = async () => {
+    try {
+      await props.contentsManager.delete(file.name);
+    } catch (err) {
+      console.error('Error removing file:', err);
+    } finally {
+      props.onRemove(file.id);
+    }
+  };
+
+  const fileIcon = getFileIcon(file.name, file.type);
+  return (
+    <div className="je-FileThumbnail">
+      <div className="je-FileThumbnail-preview">
+        <div className="je-FileThumbnail-icon">
+          <fileIcon.react />
+        </div>
+      </div>
+      <div className="je-FileThumbnail-info">
+        <div className="je-FileThumbnail-name" title={file.name}>
+          {file.name}
+        </div>
+        <div className="je-FileThumbnail-size">{(file.size / 1024).toFixed(1)} KB</div>
+      </div>
+      <button className="je-FileThumbnail-remove" onClick={handleRemove}>
```

Comment:
I like how this button looks and shows up on hover:

![image](https://github.com/user-attachments/assets/326c2a1d-b9b5-4c2e-9133-c599b42d6fc4)

Again, I do not see this on the mockup, did we  check with Ivonne?

Reply:
This button does not get shown on `:focus` state.  When as student with accessibility needs (unable to use mouse) uses <kbd>Tab</kbd> they can accidentally land on this button and by pressing enter inadvertently delete the file. The button is not

Reply:
This has been removed for now; keeping the conversation unresolved.